src/database/mod.rs

```diff
@@ -243,6 +244,29 @@ impl Database {
             })
             .collect::<Vec<_>>())
     }
+
+    pub async fn get_root_timestamp(
+        &self,
+        root: &Hash,
+    ) -> Result<Option<chrono::DateTime<Utc>>, Error> {
+        let query =
+            sqlx::query(r#"SELECT seen_at FROM root_history WHERE root = $1 LIMIT 1;"#).bind(root);
+        let row = self.pool.fetch_optional(query).await?;
+        Ok(row.map(|r| r.get::<_, _>(0)))
+    }
+
+    pub async fn store_root_state(&self, root: &Hash) -> Result<(), Error> {
+        let query = sqlx::query(
+            r#"INSERT INTO root_history(root, seen_at)
+                    VALUES ($1,  CURRENT_TIMESTAMP)
+                    ON CONFLICT (root)
```

Comment:
This should `ON CONFLICT DO NOTHING`, makes no domain sense to bump the root timeout here.